java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -156,23 +150,12 @@
   }
 
   /**
-   * Authenticates with a username and password.
```

Comment:
I don't see any need to remove existing auth methods.

Reply:
I renamed authenticate() to handshake() now, because authentication is now supplied as a CallOption. I brought back authenticateBasic() and also now have it return a CallOption containing the bearer token credentials if the server sent it back.

Reply:
I've restored the old authentication code and moved all new authentication APIs to org.apache.arrow.flight.auth2. I've also restored old authentication tests as well as made a variant of TestBasicAuth for testing Auth2 APIs.